drivers/video/video_mcux_mipi_csi2rx.c

```diff
@@ -0,0 +1,217 @@
+			240,
+			30,
+			0x1F,
+		},
+		{
+			320,
+			240,
+			15,
+			0x24,
+		},
+	};
+
+	csi2rxConfig.laneNum = DEFAULT_MIPI_CSI_NUM_LANES;
+
+	for (i = 0; i < ARRAY_SIZE(csi2rxHsSettle); i++) {
+		if ((fmt->width == csi2rxHsSettle[i][0]) && (fmt->height == csi2rxHsSettle[i][1]) &&
+		    (DEFAULT_CAMERA_FRAME_RATE == csi2rxHsSettle[i][2])) {
+			csi2rxConfig.tHsSettle_EscClk = csi2rxHsSettle[i][3];
+			break;
+		}
+	}
+
+	if (i == ARRAY_SIZE(csi2rxHsSettle)) {
+		LOG_ERR("Unsupported resolution");
+		return -ENOTSUP;
+	}
+
+	drv_data->csi2rxConfig = csi2rxConfig;
+
+	if (video_set_format(config->sensor_dev, ep, fmt)) {
+		return -EIO;
+	}
+
+	return 0;
+}
+
+static int mipi_csi2rx_get_fmt(const struct device *dev, enum video_endpoint_id ep,
+			       struct video_format *fmt)
+{
+	const struct mipi_csi2rx_config *config = dev->config;
+
+	if (fmt == NULL || ep != VIDEO_EP_OUT) {
+		return -EINVAL;
+	}
+
+	if (video_get_format(config->sensor_dev, ep, fmt)) {
+		return -EIO;
+	}
+
+	return 0;
+}
+
+static int mipi_csi2rx_stream_start(const struct device *dev)
+{
+	const struct mipi_csi2rx_config *config = dev->config;
+	struct mipi_csi2rx_data *drv_data = dev->data;
+
+	CSI2RX_Init((MIPI_CSI2RX_Type *)config->base, &drv_data->csi2rxConfig);
+
+	if (video_stream_start(config->sensor_dev)) {
+		return -EIO;
+	}
+
+	return 0;
+}
+
+static int mipi_csi2rx_stream_stop(const struct device *dev)
+{
+	const struct mipi_csi2rx_config *config = dev->config;
+
+	if (video_stream_stop(config->sensor_dev)) {
+		return -EIO;
+	}
+
+	CSI2RX_Deinit((MIPI_CSI2RX_Type *)config->base);
+
+	return 0;
+}
+
+static int mipi_csi2rx_get_caps(const struct device *dev, enum video_endpoint_id ep,
+				struct video_caps *caps)
+{
+	const struct mipi_csi2rx_config *config = dev->config;
+
+	if (ep != VIDEO_EP_OUT) {
+		return -EINVAL;
+	}
+
+	/* Just forward to sensor dev for now */
+	return video_get_caps(config->sensor_dev, ep, caps);
+}
+
+static const struct video_driver_api mipi_csi2rx_driver_api = {
+	.get_caps = mipi_csi2rx_get_caps,
+	.get_format = mipi_csi2rx_get_fmt,
+	.set_format = mipi_csi2rx_set_fmt,
+	.stream_start = mipi_csi2rx_stream_start,
+	.stream_stop = mipi_csi2rx_stream_stop,
+};
+
+static const struct mipi_csi2rx_config mipi_csi2rx_config_0 = {
+	.base = (MIPI_CSI2RX_Type *)DT_INST_REG_ADDR(0),
+	.sensor_dev = DEVICE_DT_GET(DT_INST_PHANDLE(0, sensor)),
+};
+
+static struct mipi_csi2rx_data mipi_csi2rx_data_0;
+
+static int mipi_csi2rx_init_0(const struct device *dev)
+{
+	const struct mipi_csi2rx_config *config = dev->config;
+
+	/* Check if there is any sensor device */
+	if (!device_is_ready(config->sensor_dev)) {
+		return -ENODEV;
+	}
+
+	/* MIPI DPHY power on and isolation off */
+#if defined(CONFIG_SOC_SERIES_IMXRT11XX)
+	PGMC_BPC4->BPC_POWER_CTRL |= (PGMC_BPC_BPC_POWER_CTRL_PSW_ON_SOFT_MASK |
+				      PGMC_BPC_BPC_POWER_CTRL_ISO_OFF_SOFT_MASK);
+#endif
+
+	return 0;
+}
+
+DEVICE_DT_INST_DEFINE(0, &mipi_csi2rx_init_0, NULL, &mipi_csi2rx_data_0,
```

Comment:
Why not make this driver multi-instance? It seems like we've done all the legwork of avoiding global static variables, so initializing this driver in a multi instance macro wouldn't be a lot more effort (and would make it possible to support multiple CSI2RX peripherals if we chose to implement that on an SOC in the future)

Reply:
Yes, done. I just want to add that in the "future", even if we want to support multiple camera sensors at the same time, it does not necessarily to replicate the CSI2RX. Except for very simple IPs, we could do the replication.
But in general, we don't do that to be able to reuse things (registers, for example), so multi-instance driver seems not needed for this case, I think.